### PR TITLE
Add option to allow users to control the placement of labels along line features

### PR DESCRIPTION
**Pull request by nyalldawson**
The new "Label Anchoring" section in the line placement settings for
labels allows users to specify whether labels should be placed
at the center, start or end of lines (or a custom percent from the
start of the line). Also allows data-defined percent along line
control!

Peek 2020-08-17 12-28

### CODE REVIEW

File: images/images.qrc [hunk 1]
```diff
@@ -885,6 +885,10 @@
         <file>themes/default/mIconGoochMaterial.svg</file>
         <file>themes/default/mIconSimpleLineMaterial.svg</file>
         <file>themes/default/mIconPhongTexturedMaterial.svg</file>
+        <file>themes/default/mActionLabelAnchorCenter.svg</file>
```
Comment (nirvn) [inline images/images.qrc:888]: Really good job on these icons, they are looking great and easy to understand.

File: src/ui/qgstextformatwidgetbase.ui [hunk 2]
```diff
@@ -5460,6 +5460,32 @@ font-style: italic;</string>
...
                        </widget>
                       </item>
+                      <item>
+                       <widget class="QgsCollapsibleGroupBox" name="mLineAnchorGroupBox">
+                        <property name="title">
+                         <string>Label Anchoring</string>
+                        </property>
+                        <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,0">
+                         <item>
+                          <widget class="QLabel" name="mLblNoObstacle_2">
```
Comment (nirvn) [inline src/ui/qgstextformatwidgetbase.ui:5470]:
```suggestion
                          <widget class="QLabel" name="mLblPlacementAlongLine">
```

**Review by nirvn — APPROVED** [review]
Nice!

File: images/images.qrc (hunk 1 above)
Comment (nyalldawson) [reply images/images.qrc:888]: Thanks! I'd like to see us add more little helper icons like this in future -- they really make it easier for new users to instantly understand settings.

**Pull request merged**